From 19d610a74448ee396c8df33c9696a9d5f150505f Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Mon, 26 Oct 2009 13:14:48 -0500
Subject: [PATCH 1/8] Add extra TI error codes for debugging
To: gst_omap4@list.ti.com

In theory, none of these error codes should be returned to the IL client.  But these are here for debugging.
---
 omx/gstomx_util.c |   17 +++++++++++++++++
 1 files changed, 17 insertions(+), 0 deletions(-)

diff --git a/omx/gstomx_util.c b/omx/gstomx_util.c
index 326ae83..5122442 100644
--- a/omx/gstomx_util.c
+++ b/omx/gstomx_util.c
@@ -26,6 +26,12 @@
 
 #include "gstomx.h"
 
+
+#ifdef USE_OMXTICORE
+#  include <OMX_TI_Index.h>
+#endif
+
+
 GST_DEBUG_CATEGORY (gstomx_util_debug);
 
 
@@ -286,6 +292,17 @@ g_omx_error_to_str (OMX_ERRORTYPE omx_error)
         case OMX_ErrorTunnelingUnsupported:
             return "Tunneling is unsupported by the component";
 
+#ifdef USE_OMXTICORE
+        /* note: these following warning codes should *not* be returned to the
+         * OpenMAX IL client.. but they are included here just in case
+         */
+        case OMX_TI_WarningAttributePending:
+            return "Warning: Control attribute pending";
+        case OMX_TI_WarningInsufficientAttributeSize:
+            return "Warning: Insufficient attribute size";
+        case OMX_TI_WarningEosReceived:
+            return "Warning: Eos Received";
+#endif
         default:
             return "Unknown error";
     }
-- 
1.6.3.2

